common/app/views/support/HtmlCleaner.scala

```diff
@@ -630,6 +630,46 @@ object setSvgClasses {
   }
 }
 
+case class CommercialMPUForFronts(isNetworkFront: Boolean) extends HtmlCleaner {
+  override def clean(document: Document): Document = {
+
+    def isNetworkFrontWithThrasher(element: Element, index: Int): Boolean = {
+      index == 0 && isNetworkFront && element.hasClass("fc-container--thrasher")
+    }
+
+    def hasAdjacentCommercialContainer(element: Element): Boolean = {
+      val maybeNextEl: Option[Element] = Option(element.nextElementSibling())
+      element.hasClass("fc-container--commercial") || maybeNextEl.exists(_.hasClass("fc-container--commercial"))
+    }
+
+    val sliceSlot = views.html.fragments.items.facia_cards.sliceSlot
+
+    val containers: List[Element] = document.getElementsByClass("fc-container").toList
+
+    // On mobile, we remove the first container if it is a thrasher on a Network Front
+    // and remove a container if it, or the next sibling, is a commercial container
+    // then we take every other container, up to a maximum of 10, for targeting MPU insertion
+    val containersForCommercialMPUs = containers.zipWithIndex.collect {
+      case (x, i) if !isNetworkFrontWithThrasher(x, i) && !hasAdjacentCommercialContainer(x) => x
+    }.zipWithIndex.collect {
+      case (x, i) if i % 2 == 0 => x
+    }.take(10)
+
+    for (container <- containersForCommercialMPUs) {
+      container.after(s"""<section class="fc-slice__item--mpu-candidate--mobile">${sliceSlot(containersForCommercialMPUs.indexOf(container), isMobile = true)}</section>""")
```

Comment:
I'm using the class as part of the testing, to ensure the sections are being inserted at the correct places.  Will change it to `fc-container__mpu--mobile` 👍